### DISCUSSION TITLE
c# support Nullable Reference Types

### DISCUSSION THREAD

**Discussion by ericsampson** [zeroc-ice/ice#1486] [Ideas]
I just started looking at ice for use with C#, but I'm disappointed in the custom Optional type instead of supporting Nullable Reference Types (aka NRT). That would be way more ergonomic and idiomatic.

Is this something that has been/is being considered for implementation?

Thanks
Eric

**Comment by bernardnormier** [comment]
Hi Eric,

We agree! It would be much nicer to use nullable references (and value types) in C# as opposed to some proprietary helper.

We'll fix this very soon... stay tuned.

Cheers,
Bernard

**Comment by bernardnormier** [comment]
Hi Eric,

This is now implemented!

The new Slice language—released together with IceRPC—supports optional types:
https://docs.icerpc.dev/slice2/language-guide/fields#optional-type

Any Slice type can be marked optional (with a `?` suffix) when you use it for a field or parameter. For example `string?`.

This syntax also works in Slice1 mode (= Ice compat mode) but only for "nullable" types in Ice (class references, proxies, tagged fields/parameters).

This new Slice `?` is naturally mapped to `?` in C#.

Cheers,
Bernard
